src/map/pc.c

```diff
@@ -534,6 +535,17 @@ void pc_rental_expire(struct map_session_data *sd, int i) {
 			break;
 	}
 
+	pc->setreg(sd, script->add_str("@expire_itemid"), nameid);
+	pc->setreg(sd, script->add_str("@expire_refine"), sd->status.inventory[i].refine);
+	pc->setreg(sd, script->add_str("@expire_identify"), sd->status.inventory[i].identify);
+	pc->setreg(sd, script->add_str("@expire_attribute"), sd->status.inventory[i].attribute);
+	pc->setreg(sd, script->add_str("@expire_bound"), sd->status.inventory[i].bound);
+	// pc->setreg(sd, script->add_str("@expire_uniqueid"), sd->status.inventory[i].unique_id);
```

Comment:
It doesn't convert it's like lying on the compiler and force it to compile.
The lowest unique number is already higher than int 32

On 20:21, Tue, 29 Dec 2015 AnnieRuru <notifications@github.com> wrote:

> In src/map/pc.c
> <https://github.com/HerculesWS/Hercules/pull/1046#discussion_r48559058>:
>
> > @@ -534,6 +535,17 @@ void pc_rental_expire(struct map_session_data *sd, int i) {
> >  			break;
> >  	}
> >
> > +	pc->setreg(sd, script->add_str("@expire_itemid"), nameid);
> > +	pc->setreg(sd, script->add_str("@expire_refine"), sd->status.inventory[i].refine);
> > +	pc->setreg(sd, script->add_str("@expire_identify"), sd->status.inventory[i].identify);
> > +	pc->setreg(sd, script->add_str("@expire_attribute"), sd->status.inventory[i].attribute);
> > +	pc->setreg(sd, script->add_str("@expire_bound"), sd->status.inventory[i].bound);
> > +	// pc->setreg(sd, script->add_str("@expire_uniqueid"), sd->status.inventory[i].unique_id);
>
> pc->setreg(sd, script->add_str("@expire_uniqueid"),
> (int)sd->status.inventory[i].unique_id);
> just add (int) can convert uint64 into int
> but yes, will data lost if it overflow the value
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HerculesWS/Hercules/pull/1046/files#r48559058>.
>